jsonnet/kube-thanos/kube-thanos-query.libsonnet

```diff
@@ -0,0 +1,164 @@
+local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
+
+{
+  local tq = self,
+
+  config:: {
+    name: error 'must provide name',
+    namespace: error 'must provide namespace',
+    version: error 'must provide version',
+    image: error 'must provide image',
+    replicas: error 'must provide replicas',
+    replicaLabels: error 'must provide replica labels',
+    stores: error 'must provide store addresses',
```

Comment:
Having to always pass addresses kind of blocks the use case where one only wants to run the sidecar with a Querier.

Reply:
Passing `[]` doesn't generate the store endpoint flags. This might be a good candidate for falling back to `[]` though.

Reply:
Not sure I understand. Why would it not need to be explicit when using sidecars? We would still want that to be explicit or am I missing something?

Reply:
That's true. I guess I was thinking about something that's not really a use case then :man_shrugging: